provider/gitlab.go

```diff
@@ -0,0 +1,139 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/karlderkaefer/cdk-notifier/config"
+	"github.com/xanzy/go-gitlab"
+)
+
+// NotesService interface for required Gitlab actions with API
+type GitlabNotesService interface {
+	ListMergeRequestNotes(pid interface{}, mergeRequest int, opt *gitlab.ListMergeRequestNotesOptions, options ...gitlab.RequestOptionFunc) ([]*gitlab.Note, *gitlab.Response, error)
+	DeleteMergeRequestNote(pid interface{}, mergeRequest, note int, options ...gitlab.RequestOptionFunc) (*gitlab.Response, error)
+	UpdateMergeRequestNote(pid interface{}, mergeRequest, note int, opt *gitlab.UpdateMergeRequestNoteOptions, options ...gitlab.RequestOptionFunc) (*gitlab.Note, *gitlab.Response, error)
+	CreateMergeRequestNote(pid interface{}, mergeRequest int, opt *gitlab.CreateMergeRequestNoteOptions, options ...gitlab.RequestOptionFunc) (*gitlab.Note, *gitlab.Response, error)
+}
+
+// ProjectsService interface for required Gitlab actions with API
+type GitlabProjectsService interface {
+	GetProject(pid interface{}, opt *gitlab.GetProjectOptions, options ...gitlab.RequestOptionFunc) (*gitlab.Project, *gitlab.Response, error)
+}
+
+// GitlabClient GitLab client configuration
+type GitlabClient struct {
+	Notes       GitlabNotesService
+	Projects    GitlabProjectsService
+	Context     context.Context
+	Client      *gitlab.Client
+	Config      config.NotifierConfig
+	ProjectId   string
+	NoteContent string
+}
+
+func NewGitlabClient(ctx context.Context, config config.NotifierConfig) *GitlabClient {
+	c := &GitlabClient{
+		Config:  config,
+		Context: ctx,
+	}
+	if ctx == nil {
+		c.Context = context.Background()
+	}
+
+	c.Client, _ = gitlab.NewClient(config.Token, gitlab.WithBaseURL(config.Url))
+
+	if c.Notes == nil {
+		c.Notes = c.Client.Notes
+	}
+
+	if c.Projects == nil {
+		c.Projects = c.Client.Projects
+	}
+
+	return c
+}
+
+func convert(n *gitlab.Note) *Comment {
+	var comment = &Comment{}
+	if n == nil {
+		return comment
+	}
+
+	comment.Id = int64(n.ID)
+	comment.Body = n.Body
+
+	return comment
+}
+
+func (gc *GitlabClient) GetProjectId() (string, error) {
+	if gc.ProjectId != "" {
+		return gc.ProjectId, nil
+	}
+
+	project, _, err := gc.Projects.GetProject(gc.Config.RepoOwner+"/"+gc.Config.RepoName, nil)
+
+	if err == nil {
+		return fmt.Sprint(project.ID), nil
+	}
+
+	return gc.ProjectId, err
```

Comment:
just for style I would like to return error earlier
```suggestion
	if err != nil {
		return "", err
	}

	return gc.ProjectId, nil
```